ChessVariantsLogic/Movement/RegularPattern.cs

```diff
@@ -0,0 +1,33 @@
+namespace ChessVariantsLogic;
+
+/// <summary>
+/// Represent a movement pattern with dynamic range, which DOES NOT allow jumping over occupied squares. Implements IPattern.
+/// </summary>
+public class RegularPattern : IPattern
+{
+    private readonly Tuple<int, int, int, int> pattern;
+
+    public RegularPattern(int xDir, int yDir, int minLength, int maxLength)
+    {
+        this.pattern = new Tuple<int,int,int,int>(xDir, yDir, minLength, maxLength);
+    }
```

Comment:
It's only possible to figure out what each int in the `pattern` tuple represents by reading the constructor or the methods. Since the tuple itself is private now, wouldn't it make more sense to save these as separate variables, for clarity? i.e:

int xDir;
int yDir;
int minLength;
int maxLength;

Same applies for any other patterns.